extensions/wikia/PortableInfobox/PortableInfoboxHooks.class.php

```diff
@@ -10,4 +10,17 @@ static public function onBeforePageDisplay( OutputPage $out, Skin $skin ) {
 
 		return true;
 	}
+
+	static public function onImageServingCollectImages( &$imageNamesArray, $articleTitle ) {
+		if ( $articleTitle ) {
+
+			$dataService = new PortableInfoboxDataService();
+			$infoboxData = $dataService->getInfoboxDataByTitle( $articleTitle );
+			$infoboxImages = $dataService->getImageListFromInfoboxData( $infoboxData );
+			if ( !empty( $infoboxImages ) ) {
```

Comment:
Can $imageNamesArray be a non-array? If so, the below array_merge needs to support it to avoid notice flood